### ISSUE TITLE
[Supportal] Make Supportal debug console mirror user's console

### ISSUE THREAD

**Issue by stephanieelliott** [Expensify/App#51632]
### Context: 

To help support customers in New Expensify, we need to ensure that when Supportaled into customer accounts, the debug tools, including Onyx data, show the actual data the user would see. This would help us diagnose RBR/GBR/unread errors we see.

**Version Number:**
**Reproducible in staging?:** Yes 
**Reproducible in production?:** Yes 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):** manager.stephanie@expensicorp.com
**Logs:** N/A
**Expensify/Expensify Issue URL:** N/A
**Issue reported by:** @stephanieelliott / @danielrvidal 
**Slack conversation:** https://expensify.slack.com/archives/C07NZ8B1VTQ/p1728500635598069

## Action Performed:
1. In browser A, Support log into a test account 
2. In browser B, directly log into the same test account 
3. In both browsers (A and B), navigate to Settings > Troubleshoot > View Debug Console 

## Expected Result:
_Note: this is an improvement, not a regression_
Debug console should be the same in browsers A and B 

## Actual Result:
Debug console in browsers A and B is not the same 

## Workaround:
N/A 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<img width="844" alt="image" src="https://github.com/user-attachments/assets/a4342010-2680-402b-b4c5-548d4dd5a79f">

## Example:

**Supportal data**
 - [onyx-state.txt](https://github.com/user-attachments/files/17553487/onyx-state.txt)
 - Debug console: [logs-2024-10-29 07_55_54.629.txt](https://github.com/user-attachments/files/17553494/logs-2024-10-29.07_55_54.629.txt)

**Direct login data** 
 - [onyx-state-2.txt](https://github.com/user-attachments/files/17553514/onyx-state-2.txt)
 - Debug console: [logs-2024-10-29 07_57_12.883.txt](https://github.com/user-attachments/files/17553510/logs-2024-10-29.07_57_12.883.txt)

<details><summary>Issue Owner</summary>Current Issue Owner: @youssef-lr</details>

**Comment by dannymcclain** [comment]
Doesn't seem like this actually needs any design support—mostly just code wizardry yeah? 🧙

**Comment by RachCHopkins** [comment]
@maddylewis should I add the `hot picks` label to this?

**Comment by youssef-lr** [comment]
@stephanieelliott @danielrvidal have we confirmed this is an issue? Because I think Onyx logs will just print whatever the app has done and it should technically be the same. Have we confirmed that by taking Action A, different logs were printed in Supportal vs login? if not I can conduct this test.

**Comment by RachCHopkins** [comment]
@youssef-lr I think it's worth testing that - I'm happy to help, if you can spoon-feed me some instructions!

**Comment by stephanieelliott** [comment]
@youssef-lr this was tested -- see the end of the issue body!

The image at end shows the debug console viewed from the support login vs direct login, then there are attached logs for for each login (supportal vs direct), as well as a TXT file of the Onyx logs for each login.

**Comment by youssef-lr** [comment]
Gonna get to this today

**Comment by VictoriaExpensify** [comment]
Any update on this one @youssef-lr ?

**Comment by youssef-lr** [comment]
I haven't been able to find out if there is anything that could make these logs different. Onyx logging is simply whatever the app is doing being logged, and it's pretty low level so I don't see how Supportal could affect it. If logs are different then it should be related to just how Supportal might be doing some things differently, but it shouldn't affect whether or not we can get accurate Onyx regarding a bug a customer is experiencing. 

In regards to the logs being different here in this issue, logs seems to sometimes persist from one login to another, but in my tests I made sure I start with fresh logs and perform one action and compare. For this one the debug console was showing just one line when I first started the test, then I navigated to the home page and then back to the debug console, and the logs seem to be pretty similar. They can't be 100% the same because Pusher updates can happen at different times and mostly timing of things affect how logs look like.

So in conclusion I think we can close this.

[logs_direct.txt](https://github.com/user-attachments/files/18085335/logs_direct.txt)


[logs_supportal.txt](https://github.com/user-attachments/files/18085334/logs_supportal.txt)
